microsoft/vscode
# Fix tasks status incorrect #163393

Fixes #163393

## How to Test

Open vscode with source with `tasks.json` defined like below, tasks status should be correct after task is done

(Use `cmd+shift+b` to start task)

```json
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "echo11",
            "type": "shell",
            "command": "${workspaceFolder}/a.sh",
            "presentation": {
                "echo": true,
                "reveal": "always",
                "focus": false,
                "panel": "new",
                "showReuseMessage": true,
                "clear": false
            },
            "problemMatcher": [],
            "group": {
                "kind": "build",
                "isDefault": true
            }
        }
    ]
}
```

```json
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "echo11",
            "type": "shell",
            "command": "${workspaceFolder}/a.sh",
            "presentation": {
                "echo": true,
                "reveal": "always",
                "focus": false,
                "showReuseMessage": true,
                "clear": false
            },
            "problemMatcher": [],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "runOptions": {
                "instanceLimit": 3
            }
        }
    ]
}
```

File `a.sh` can be like
```shell
echo hello
sleep 3
```
